keel-core/src/main/kotlin/com/netflix/spinnaker/keel/pause/ActuationPauser.kt

```diff
@@ -102,10 +102,27 @@ class ActuationPauser(
     mutableListOf<PersistentEvent>()
       .let { events ->
         events.addAll(originalEvents)
-        val relevantAppEvents = resourceRepository
-          .applicationEventHistory(resource.application, events.last().timestamp)
+
+        val oldestResourceEvent = events.last()
+
+        if (oldestResourceEvent is ResourceCreated) {
+          val pause = pausedRepository.getPause(APPLICATION, resource.application)
+          if (pause != null && pause.pausedAt.isBefore(oldestResourceEvent.timestamp)) {
+            // If the app was paused before the resource was created, and hasn't yet been resumed,
+            // we add that event to the history so the resource event history makes sense against its status
+              events.add(ApplicationActuationPaused(resource.application, pause.pausedAt))
+          }
+        }
```

Comment:
This is much simplified from my previous attempt. Now that we have timestamps associated with records in the `paused` table, which are _not_ deleted upon deleting delivery configs or resources, we can synthesize paused events in a more straightforward fashion.